src/test/java/school/hei/haapi/integration/TeacherIT.java

```diff
@@ -277,6 +277,39 @@ void manager_write_update_ok() throws ApiException {
     assertEquals(toUpdate, updated.get(0));
   }
 
+  @Test
+  void manager_write_update_without_email_field_ko() throws ApiException {
+    ApiClient manager1Client = anApiClient(MANAGER1_TOKEN);
+    UsersApi api = new UsersApi(manager1Client);
+
+    List<Teacher> toUpdate =
+        api.createOrUpdateTeachers(List.of(someCreatableTeacher()));
+
+    Teacher toUpdate0 = toUpdate.get(0);
+    toUpdate0.setEmail(null);
+
+    assertThrowsApiException(
+        "{\"type\":\"400 BAD_REQUEST\",\"message\":\"Email is mandatory\"}",
+        () -> api.createOrUpdateTeachers(toUpdate));
+  }
+
+  @Test
+  void manager_write_update_without_names_ko() throws ApiException {
+    ApiClient manager1Client = anApiClient(MANAGER1_TOKEN);
+    UsersApi api = new UsersApi(manager1Client);
+
+    List<Teacher> toUpdate =
+        api.createOrUpdateTeachers(List.of(someCreatableTeacher()));
+
+    Teacher toUpdate0 = toUpdate.get(0);
+    toUpdate0.setFirstName(null);
+    toUpdate0.setLastName(null);
+
+    assertThrowsApiException(
+        "{\"type\":\"400 BAD_REQUEST\",\"message\":\"Some fields are still empty\"}",
```

Comment:
You have to tell which ones

Reply:
Initially my idea, but after thinking a while, just for now, we already have 8 fields in our user form. So should we display all these 8 fields if they are still blank ?